x-pack/plugins/metrics_entities/server/modules/types.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * Loose type for the mappings
+ */
+export interface Mappings {
+  [key: string]: unknown;
+  mappings: {
+    [key: string]: unknown;
+    _meta: {
+      index: string;
+    };
+  };
+}
+
+/**
+ * Loose type for the transforms. id is marked optional so we can delete it before
+ * pushing it through elastic client.
+ * TODO: Can we use stricter pre-defined typings for the transforms here or is this ours because we define it slightly different?
```

Comment:
++, are there typings directly from the Transforms plugin that can be used?

Reply:
That's a good question. I will look into that and see if there are. I am more hopeful there are es types for them and eventually use those instead. However, the way I use transforms is a very limited subset, so I still might have to make my own types or use a Partial<> with the types.